react/features/remote-video-menu/components/RemoteVideoMenuTriggerButton.js

```diff
@@ -162,32 +168,52 @@ class RemoteVideoMenuTriggerButton extends Component {
             participantID
         } = this.props;
 
-        return (
-            <RemoteVideoMenu id = { participantID }>
-                { isModerator
-                    ? <MuteButton
-                        isAudioMuted = { isAudioMuted }
-                        onClick = { this._onRemoteMenuClose }
-                        participantID = { participantID } />
-                    : null }
-                { isModerator
-                    ? <KickButton
-                        onClick = { this._onRemoteMenuClose }
-                        participantID = { participantID } />
-                    : null }
-                { remoteControlState
-                    ? <RemoteControlButton
-                        onClick = { onRemoteControlToggle }
-                        participantID = { participantID }
-                        remoteControlState = { remoteControlState } />
-                    : null }
-                { onVolumeChange
-                    ? <VolumeSlider
-                        initialValue = { initialVolumeValue }
-                        onChange = { onVolumeChange } />
-                    : null }
-            </RemoteVideoMenu>
-        );
+        const buttons = [];
+
+        if (isModerator) {
+            buttons.push(
+                <MuteButton
+                    isAudioMuted = { isAudioMuted }
+                    key = 'mute'
+                    onClick = { this._onRemoteMenuClose }
+                    participantID = { participantID } />
+            );
+            buttons.push(
+                <KickButton
+                    key = 'kick'
+                    onClick = { this._onRemoteMenuClose }
+                    participantID = { participantID } />
+            );
+        }
+
+        if (remoteControlState) {
+            buttons.push(
+                <RemoteControlButton
+                    key = 'remote-control'
+                    onClick = { onRemoteControlToggle }
+                    participantID = { participantID }
+                    remoteControlState = { remoteControlState } />
+            );
+        }
+
+        if (onVolumeChange && isModerator) {
+            buttons.push(
+                <VolumeSlider
+                    initialValue = { initialVolumeValue }
+                    key = 'volume-slider'
+                    onChange = { onVolumeChange } />
+            );
+        }
+
+        if (buttons.length > 0) {
+            return (
+                <RemoteVideoMenu id = { participantID }>
+                    { buttons }
+                </RemoteVideoMenu>
+            );
+        }
+
+        return null;
```

Comment:
Thanks for fixing the conflicts. After filmstrip react-ification/redux-ification is complete I plan on doing another pass at the components for relational/architectural cleanup so maybe then I can work on moving out this logic from the trigger.